pdns/recursordist/aggressive_nsec.cc

```diff
@@ -580,6 +580,18 @@ bool AggressiveNSECCache::getNSEC3Denial(time_t now, std::shared_ptr<AggressiveN
         break;
       }
 
+      const DNSName signer = getSigner(closestNSEC3.d_signatures);
+      if (type != QType::DS && isNSEC3AncestorDelegation(signer, closestNSEC3.d_owner, nsec3)) {
```

Comment:
Does this mean we do allow the NSEC3 from above the cut to deny a DS one level below the cut?

Reply:
Nice catch! I pushed a fix, several comments, and amended the test to catch that as well.